### ISSUE TITLE
Scope and Requirements

### ISSUE THREAD

**Issue by apoliveira** [apoliveira/software-engineering#3]
Here's the scope first draft for the scope and requirements. Let me know what you think @nymim93  @tmwhite 

**Problem**: It is common to here presenters complain about the hassle of giving a presentation. Even before giving the presentation they must get the presentation in a nice format and onto a specific computer. Then, during the presentation they are typically tied down to the podium/central computer to navigate through their presentation. The audience also suffers in certain cases where it's difficult to see the screen and or hear the presenter. A natural occurrence of this dilemma is for teachers and presenters.

**Scope**: Black Hawk will mitigate these issues by creating a website to host one's presentations. Delivering a presentation will be a matter of logging in to the website and beginning. A presenter is then able to use any Internet connected device as a remote to control the presentation. The presenter is also able to draw on top of the presentation screen. Not only will these actions (previous/next page, drawings) be transmitted to the screen, but the audience members are able to follow the presentation on their own devices. 
Outside of the classroom there are office hours. These meetings between teacher and student usually require a collaborative writing area. Along with the drawing feature allowed for presentations, teachers will be able to have virtual white-board sessions allowing for interaction between teachers and students.

**Stake Holders**: The stakeholders are divided into 2 groups: Dr. Andrew Greene (the client), and the developers (as our grades are on the line).

**Risk Analysis**: Our product will be freely hosted on GitHub and Heroku, thus incurring no cost on any of the stakeholders. While there is a minor risk of the presentations not broadcasting correctly to other devices, the chances the website will go down are minuscule. So, a teacher can at the least download his/her presentation and continue accordingly. The largest risk for the project is the loss of time and sleep on the developer end, as well as a deep sadness for the client.

**Requirements**:
- Presentations
  - Presenter
    - Upload list of presentations (pdf)
    - Have a user page showcasing presentations
    - Start/View presentations
    - Broadcast presentation actions (previous/next page, drawing on slides, and audio)
  - Audience
    - Watch presentations – updated by the presenter's view
- Whiteboard
  - Presenter
    - Start virtual whiteboard
    - Send invitation to others
    - Draw on a virtual whiteboard
  - User/Student
    - Draw on a virtual whiteboard (after invitation)

**Comment by nymim93** [comment]
Love the risk analysis and the stake holders! But everything here looks right to me, and I'm pretty sure you covered everything.

**Comment by apoliveira** [comment]
Although it wasn't updated before here is the final scope and requirements that we have agreed on, and have been approved.

##Problem

- Giving presentations are hard
- Making sure the presentation will display/work correctly is an extra hassle
- Presenters are tied down to the podium
- Audience members have difficulty seeing and hearing the presentation

##The Fix/Scope

- Create a website to host these presentations
- Allow presenter to use any browser-equipped device to control the presentation
- Allow presenter to draw on presentation
- Broadcast presentation, in real time, to the audience member’s devices
- So that audience members can more easily view the presentation
- Also allows viewing by individuals not physically present

#Requirements

##Presentation

###Presenter

- Upload presentations (pdf)
- Have a user page showcasing presentations
- Start/View Presentations
- Broadcast presentation actions (in real time)
- Previous/Next page
- Draw on slide
- Audio

###Audience

- Watch presentations 
  - updated by the presenter’s view in real
time

##Whiteboard

###Presenter

- Start virtual whiteboard
- Send invitation to others
- Draw on a virtual whiteboard

###User/Student

- Draw on a virtual whiteboard
- Only after invitation

#Risks

##Stakeholders

- The client, Dr. Andrew Greene, has his hopes and dreams at stake.
- The developers have their grades on the line. . .

##Risk Analysis

###Low Priority Risks

- The project will not be used by anyone
  - Possibility: Unsure
  - Effect: Application sits idle
  - Remedy: There’s no need to fix this

###Medium Priority Risks

- GitHub or Heroku will go out of business
  - Possibility: Unlikely
  - Effect: Our code and website would go down
  - Remedy: Any disruption of service can be remedied by pushing to another cloud-based platform

###High Priority Risks

- The project is used, but is malfunctional
  - Possibility: Unsure
  - Effect: A presenter gives a bad presentation, ruins their reputation
  - Remedy: None besides write good code and have hard reviews
- Will not be able to complete barebones application
  - Possibility: Unlikely
  - Effect: We do badly in this course and the client is very upset. . .
  - Remedy: We’re in the process of prototyping each component of the application

- Application will not be made to client’s satisfaction
  - Possibility: Unlikely
  - Effect: We do badly in this course and the client is very upset. . . (again)
  - Remedy: Use an evolutionary process flow and collaborating with the client with each evolution.

**Comment by apoliveira** [comment]
Although it wasn't updated before here is the final scope and requirements that we have agreed on, and have been approved.

##Problem
- Giving presentations are hard
- Making sure the presentation will display/work correctly is an extra hassle
- Presenters are tied down to the podium
- Audience members have difficulty seeing and hearing the presentation

##The Fix/Scope
- Create a website to host these presentations
- Allow presenter to use any browser-equipped device to control the presentation
- Allow presenter to draw on presentation
- Broadcast presentation, in real time, to the audience member’s devices
- So that audience members can more easily view the presentation
- Also allows viewing by individuals not physically present

#Requirements

##Presentation

###Presenter
- Upload presentations (pdf)
- Have a user page showcasing presentations
- Start/View Presentations
- Broadcast presentation actions (in real time)
- Previous/Next page
- Draw on slide
- Audio

###Audience
- Watch presentations 
  - updated by the presenter’s view in real
    time

##Whiteboard

###Presenter
- Start virtual whiteboard
- Send invitation to others
- Draw on a virtual whiteboard

###User/Student
- Draw on a virtual whiteboard
- Only after invitation

#Risks

##Stakeholders
- The client, Dr. Andrew Greene, has his hopes and dreams at stake.
- The developers have their grades on the line. . .

##Risk Analysis

###Low Priority Risks
- The project will not be used by anyone
  - Possibility: Unsure
  - Effect: Application sits idle
  - Remedy: There’s no need to fix this

###Medium Priority Risks
- GitHub or Heroku will go out of business
  - Possibility: Unlikely
  - Effect: Our code and website would go down
  - Remedy: Any disruption of service can be remedied by pushing to another cloud-based platform

###High Priority Risks
- The project is used, but is malfunctional
  - Possibility: Unsure
  - Effect: A presenter gives a bad presentation, ruins their reputation
  - Remedy: None besides write good code and have hard reviews
- Will not be able to complete barebones application
  - Possibility: Unlikely
  - Effect: We do badly in this course and the client is very upset. . .
  - Remedy: We’re in the process of prototyping each component of the application
- Application will not be made to client’s satisfaction
  - Possibility: Unlikely
  - Effect: We do badly in this course and the client is very upset. . . (again)
  - Remedy: Use an evolutionary process flow and collaborating with the client with each evolution.
